### PR TITLE
Add time offset for LowStorageIntegrator

### PR DESCRIPTION
**Pull request by pbrady**
## PR Summary

I was wanting to supply a time dependent source term (for manufactured solutions) using the LowStorageIntegrator and didn't see a direct way to access the time level associated with each stage.  I followed the procedure in [Ketcheson 2010](https://doi.org/10.1016/j.jcp.2009.11.006) to compute the Butcher table coefficients for the methods.  The resulting `c`s were tested out in a Mathematica notebook and recovered the expected order of accuracy.  

## PR Checklist

- [ ] Code passes cpplint
- [x] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [ ] Change is breaking (API, behavior, ...)
  - [ ] Change is *additionally* added to CHANGELOG.md in the breaking section
  - [ ] PR is marked as breaking
  - [ ] Short summary API changes at the top of the PR (plus optionally with an automated update/fix script)
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [ ] Docs build
- [ ] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Review by Yurlungur — APPROVED** [review]
Nice improvement!

**Comment by pbrady** [comment]
Seems like a CI failure: 
```console
Error: Could not find file '/home/runner/work/_actions/_temp_68ccd911-c253-4309-82e1-094e899b9ceb/_staging/gregsdennis-dependencies-action-f4fa194/node_modules/.bin/esparse'.
```

**Comment by pgrete** [comment]
> Seems like a CI failure:
> 
> ```
> Error: Could not find file '/home/runner/work/_actions/_temp_68ccd911-c253-4309-82e1-094e899b9ceb/_staging/gregsdennis-dependencies-action-f4fa194/node_modules/.bin/esparse'.
> ```

Yes, somehow the automated docker image cleanup didn't work as expected. I'm looking into it.

**Comment by pgrete** [comment]
> > Seems like a CI failure:
> > ```
> > Error: Could not find file '/home/runner/work/_actions/_temp_68ccd911-c253-4309-82e1-094e899b9ceb/_staging/gregsdennis-dependencies-action-f4fa194/node_modules/.bin/esparse'.
> > ```
> 
> Yes, somehow the automated docker image cleanup didn't work as expected. I'm looking into it.

Things should work as before again.

File: src/time_integration/low_storage_integrator.cpp [hunk 1]
```diff
@@ -128,31 +140,38 @@ LowStorageIntegrator::LowStorageIntegrator(const std::string &name)
...
     beta[1] = 2.364680399061355;
     gam0[1] = -3.666545952121251;
     gam1[1] = 4.666545952121251;
+    c[1] = 0.357534921136978;
 
     delta[2] = 0.0;
     beta[2] = 0.016239790859612;
     gam0[2] = 0.035802535958088;
     gam1[2] = 0.964197464041912;
+    c[2] = 1.0537621812245777;
```
Comment (pgrete) [inline src/time_integration/low_storage_integrator.cpp:161]: Is a time offset >1 allowed?

**Review by pgrete — COMMENTED** [review]
Thanks for the addition.
Would you mind adding to the doc (https://parthenon-hpc-lab.github.io/parthenon/develop/src/integrators.html) that c is now also available for the low storage integrators (and potentially even the conversion formula you used to calculate c)?
Also a changelog entry would be great.

File: src/time_integration/low_storage_integrator.cpp (hunk 1 above)
Comment (pbrady) [reply src/time_integration/low_storage_integrator.cpp:161]: That is unexpected.  I observed the expected order of convergence so I assumed it was correct but am not an expert in SSP RK methods.

Comment (pbrady) [reply src/time_integration/low_storage_integrator.cpp:161]: I couldn't find any restrictions `c` so I experimented with changing `c[2] <- 1.0` and the result was a loss of accuracy (to 1st order).

**Comment by pbrady** [comment]
I've updated the docs and changelog.  Computing `c` is more of an algorithm than a formula so I referenced the paper instead of repeating the algorithm.

**Comment by Yurlungur** [comment]
CI failing. @pbrady will plan to add mathematica notebook.

**Comment by pbrady** [comment]
Mathematica notebook added

**Comment by pgrete** [comment]
I removed the binary parts (mostly the rendered images) from the notebook file as they don't render on Github anyway (and added one sentence pointing to the notebook in both the docs in head file). Any objections?

**Comment by pbrady** [comment]
No objections here.

**Review by pgrete — APPROVED** [review]

**Pull request merged**